aten/src/ATen/native/cudnn/MHA.cpp

```diff
@@ -614,9 +664,44 @@ void run_cudnn_SDP_bprop(
     Tensor& dV,
     const Tensor& dropoutseed,
     const Tensor& dropoutoffset) {
+  if (cudnn_sdpa_check_debug()) {
+    TORCH_WARN("Calling into cuDNN SDPA BPROP");
+  }
+  Tensor dO_ = dO;
+  if (!dO.strides()[dO.strides().size() - 1]) {
+    TORCH_WARN(
+        "cuDNN SDPA backward got an innermost stride of 0 in grad_out, which is unsupported."
```

Comment:
When does this happen, I think in general people dont have control over the striding  of the backpropping grad so I dont know how actionable this warning is for a user

Reply:
This happens e.g., for some tests that do `out.sum().backward()` to exercise bprop which produces a scalar "tensor" as the grad (all ones with the same shape as `out` but strides of `[0, 0, 0, 0]`).

I'm kinda balancing the rarity of this scenario with reducing the surprise to those finding that `out.sum().backward()` incurs an additional memory allocation/fill-kernel.